lib/core/src/main/java/io/atlasmap/core/DefaultAtlasFieldActionService.java

```diff
@@ -799,7 +799,7 @@ private void extractFlatListValuesFromFieldGroup(AtlasInternalSession session,
                 // Collection field
                 List<Field> fields = ((FieldGroup) subField).getField();
                 if (fields != null && !fields.isEmpty()) {
-                    if (fieldGroup.getField().size() == 1 && (index == 0 || index == null)) {
+                    if (fieldGroup.getField().size() == 1 && (index == null || index == 0)) {
```

Comment:
Just trying to grok this - testing for 0 or null with the parenthetical expression shouldn't matter - how did this make a difference?

Reply:
So if index is null, the former tries to "unbox" `Integer` into primitive `int` to evaluate `index == 0`, causes NPE.

Reply:
ah - sneaky - good catch...